sandstone/client/sandstone/components/filetreedirective/tests/filetree.directive.test.js

```diff
@@ -1,208 +1,421 @@
+              'filename': 'tmp',
+              'filepath': '/tmp'
+          }
+      ];
 
-    it('should be able to get files for a folder', function(){
-      spyOn(filesystemservice, 'getFiles');
-      isolateScope.getDirContents(dirs[0], true);
-      expect(filesystemservice.getFiles).toHaveBeenCalled();
-    });
+      var files = [{
+            "filepath": "/home/user/dir1",
+            "filename": "dir1",
+            "group": "saurabh",
+            "is_accessible": true,
+            "perm": "-rw-rw-r--",
+            "perm_string": "664",
+            "size": "4.0 KiB",
+            "type": "dir"
+          }, {
+            "filepath": "/home/user/dir2",
+            "filename": "dir2",
+            "group": "root",
+            "is_accessible": false,
+            "perm": "-rw-r--r--",
+            "perm_string": "644",
+            "size": "4.0 KiB",
+            "type": "dir"
+          },
+          {
+            "filepath": "/home/user/file1",
+            "filename": "file1",
+            "group": "root",
+            "is_accessible": false,
+            "perm": "-rw-r--r--",
+            "perm_string": "644",
+            "size": "4.0 KiB",
+            "type": "file"
+          }];
 
-    it('should return the node given a filepath', function() {
-      var node = isolateScope.getNodeFromPath('/home/saurabh/dir1', isolateScope.treeData.filetreeContents);
-      // Expect node parameters to match
-      expect(node.filepath).toBe('/home/saurabh/dir1');
-      expect(node.filename).toBe('dir1');
-      expect(node.group).toBe('saurabh');
-      expect(node.is_accessible).toBeTruthy();
-      expect(node.perm).toBe('-rw-rw-r--');
-      expect(node.perm_string).toBe('664');
-      expect(node.size).toBe('4.0 KiB');
-    });
+      FilesystemService = {
+          getFolders: function(node, callback) {
+              if(node.filepath == '') {
+                  callback(rootDirs);
+              } else {
+                  callback(dirs, null, null, null, node);
+              }
+              // return dirs;
+          },
+          getFiles: function(node, callback) {
+              // return files;
+              if(node == '') {
+                  callback(rootDirs);
+              } else {
+                  callback(files, null, null, null, node);
+              }
+          },
+          createdNewDir: function() {
+              // Do something here
+          }
+      };
+      beforeEach(module('sandstone'));
+      beforeEach(module('sandstone.editor'));
+      beforeEach(module('sandstone.templates'));
+      beforeEach(module('sandstone.filetreedirective'));
 
-    it('should return null in case node doesnt exist', function(){
-      var node = isolateScope.getNodeFromPath('/home/saurabh/dir111', isolateScope.treeData.filetreeContents);
-      expect(node).not.toBeDefined();
-    });
+      beforeEach(module(function($provide) {
+          $provide.value('FilesystemService', FilesystemService);
```

Comment:
Mocking the `FilesystemService` is good for the controller unit tests, however the actual service should be utilized for the directive-level integration tests. For the integration tests, utilize the built-in mock `$httpBackend` instead of mocking out the service.